presto-parquet/src/main/java/com/facebook/presto/parquet/cache/CachingParquetMetadataSource.java

```diff
@@ -38,14 +38,26 @@ public CachingParquetMetadataSource(Cache<ParquetDataSourceId, ParquetFileMetada
     }
 
     @Override
-    public ParquetFileMetadata getParquetMetadata(ParquetDataSource parquetDataSource, long fileSize, boolean cacheable)
+    public ParquetFileMetadata getParquetMetadata(
+            ParquetDataSource parquetDataSource,
+            long fileSize,
+            boolean cacheable,
+            long modificationTime)
             throws IOException
     {
         try {
             if (cacheable) {
-                return cache.get(parquetDataSource.getId(), () -> delegate.getParquetMetadata(parquetDataSource, fileSize, cacheable));
+                ParquetFileMetadata fileMetadataCache = cache.get(
+                        parquetDataSource.getId(),
+                        () -> delegate.getParquetMetadata(parquetDataSource, fileSize, cacheable, modificationTime));
+                if (fileMetadataCache.getModificationTime() == modificationTime) {
+                    return fileMetadataCache;
+                }
+                else {
+                    cache.invalidate(parquetDataSource.getId());
+                }
```

Comment:
to be more specific of my previous comments, I think we might need add a couple of junit test cases to cover both valid and invalid cache cases.  what do you think?

Reply:
> to be more specific of my previous comments, I think we might need add a couple of junit test cases to cover both valid and invalid cache cases. what do you think?

Thanks for the advice, i will add unit tests about that.